src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,375 @@
+# How to create FedRAMP OSCAL validation constraints
+
+FedRAMP OSCAL documents are structured XML documents. The document syntax is defined by OSCAL XML Schema. However, XML Schema cannot express many semantic constraints, and an additional schema language is employed: Schematron.
+
+# Technologies used
+
+- [XML](https://www.w3.org/standards/xml/) is a structured document langauge.
+- [Namespaces in XML](https://www.w3.org/TR/xml-names/) allow element (and attribute) names to be declared in distinct namespaces
+- [OSCAL](https://pages.nist.gov/OSCAL/) is used to define the several FedRAMP document types in a structured data form. NIST provides XML Schema documents for the various OSCAL document types.
+- [Schematron](https://schematron.com/) is used to declare assertions for documents to be validated
+- [XPath](https://www.w3.org/TR/xpath-31/) is used in Schematron to define rule contexts and assertion tests
+- [XSpec](https://github.com/xspec/xspec) is used to create unit tests for Schematron assertions
+- [XSLT](https://www.w3.org/TR/xslt-30/) is used to manipulate XML documents, and is used to transpile Schematron documents into XSLT documents
+
+Use of Schematron will require familiarity with XPath for context identification and assertion tests, and XSpec for creation of unit tests.
+
+## Useful resources
+
+[xpather.com](http://xpather.com/) is a browser-based XPath exploration tool. It is not a strict XPath tool: namespaces in the XML document are ignored, and the XPath queries must not use include namespace prefixes. This design makes rapid prototyping easier, but readily copy-pasting XPath into Schematron not error-free.
```

Comment:
Not sure what your trying to say here regarding the "readilly copy-pasting ..." sentence.   Maybe change this to " prototyping easier but additional care must be taken when copy-pasting into schmatron'.